iree/turbine/kernel/ops/wave_ops.py

```diff
@@ -449,7 +449,7 @@ class CustomOp(ABC):
     _tracing_function: Optional[Callable[..., Any]] = field(default=None, init=False)
 
     @property
-    def location(self) -> Optional[FileLineColInfo]:
+    def location(self) -> Optional[Union[FileLineColInfo, StackTraceInfo]]:
```

Comment:
consider using new Python syntax: `FileLineColInfo | StackTraceInfo` instead of `Union[FileLineColInfo, StackTraceInfo]` here and at some other places